presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -115,32 +120,35 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
-        if (Varchars.isVarcharType(type)) {
+        if (REAL.equals(type)) {
+            return "float";
+        }
+        if (TIME_WITH_TIME_ZONE.equals(type)) {
+            return "time";
```

Comment:
MySQL `time` cannot hold time zone information carried by TIME_WITH_TIME_ZONE.

Also, `time` is not fully safe for TIME *WITHOUT* TIME ZONE, since the allowed range of values is very broad. It's safe for write-only from Presto though.

Reply:
The existing temporal type handling is likely wrong, but this commit is only refactoring the checks to use the type classes instead of comparing the string returned from the super method.

I've been thinking about how types should be supported for the JDBC connectors. It's probably better that we explicitly list every supported type in each connector rather than relying on default behavior in the base module (which can't be safely modified without knowing all of the usages).